.github/workflows/rust.yml

```diff
@@ -1,39 +1,48 @@
-name: Rust
+name: Rust Build and Artifact Creation
```

Comment:
nit: I'm not sure if "Build and Artifact Creation" really gives any relevant info... I'd just say "Build". Rust is obvious, this is a Rust project and artifacts are created more or less as a side effect.

Reply:
This is what AI though up :)

Reply:
Anyway, it is building and creating artifacts indeed, I suggest to remove `Rust`.

Reply:
I think artifacts are a side effect of a build. Every build will have artifacts (whether you upload them or not). I don't have a strong opinion though, so change it as you want

Reply:
So what about `Build and Artifact Upload`?

Reply:
It's not better for me, as I don't think it's necessary to include "artifacts" in the name, but I don't really care :D name it whatever you want. It was just nitpick anyway